### PR TITLE
SAK-50348 Microsoft Integration: Visual and logical improvements

### PR DESCRIPTION
**Pull request by mollerentornos**
Jira: https://sakaiproject.atlassian.net/browse/SAK-50348

This feature introduces several essential modifications to provide users with more insight into the status of Microsoft integration and to manage bindings with an improved UI/UX. Additionally, it includes changes to the logical structure by adding batch requests to enhance performance.

This development has been sponsored by Universitat de Lleida (UdL)

### CODE REVIEW

File: microsoft-integration/admin-tool/src/main/java/org/sakaiproject/microsoft/controller/AutoConfigController.java [hunk 1]
```diff
@@ -214,173 +226,259 @@ public Boolean autoConfigConfirm(
...
+							autoConfigSessionBean.addError(siteId, site.getTitle(), e.getMessage());
+						} finally {
+							autoConfigSessionBean.increaseCounter();
+						}
+					} else if (o instanceof SiteSynchronization) {
+						autoConfigSessionBean.addStatus(2, site.getTitle());
+						try {
+							handleExistingTeamBinding(autoConfigSessionBean, site, (SiteSynchronization) o, syncDateFrom, syncDateTo, credentials);
+						} catch (Exception e) {
+							autoConfigSessionBean.addError(siteId, site.getTitle(), e.getMessage());
```
Comment (jesusmmp) [inline microsoft-integration/admin-tool/src/main/java/org/sakaiproject/microsoft/controller/AutoConfigController.java:271]: Should we here put a log.error too?

File: microsoft-integration/admin-tool/src/main/java/org/sakaiproject/microsoft/controller/AutoConfigController.java [hunk 2]
```diff
@@ -214,173 +226,259 @@ public Boolean autoConfigConfirm(
...
+						continue;
+					}
+
+					Object o = map.get(siteId);
+					if (o == null) {
+						continue;
+					}
+
+					if (o instanceof String) {
+						autoConfigSessionBean.addStatus(1, site.getTitle());
```
Comment (jesusmmp) [inline microsoft-integration/admin-tool/src/main/java/org/sakaiproject/microsoft/controller/AutoConfigController.java:258]: Maybe the status: 0, 1, 2, 3, etc should be defined as constants. But is ok

File: microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftSynchronizationServiceImpl.java [hunk 3]
```diff
@@ -60,28 +51,49 @@
...
+	@Setter
+	private SessionManager sessionManager;
+
 	//used in hooks. Sometimes we need to stop listening some events
 	private Set<String> disabledGroupListeners = ConcurrentHashMap.newKeySet();
-	//used in hooks to synchronize. "add users to group" must happen after "create group" 
+	//used in hooks to synchronize. "add users to group" must happen after "create group"
 	private ConcurrentHashMap<String, Object> newGroupLock = new ConcurrentHashMap<>();
 
+	//private static final int MAX_CHANNELS = 3;
```
Comment (jesusmmp) [inline microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftSynchronizationServiceImpl.java:95]: Delete comment

**Comment by mollerentornos** [comment]
PR updated

File: microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/jobs/RunSynchronizationsJob.java [hunk 4]
```diff
@@ -54,26 +70,52 @@ public void init() {
...
 			
 			List<SiteSynchronization> list = microsoftSynchronizationService.getAllSiteSynchronizations(true);
 			for(SiteSynchronization ss : list) {
-				try {
-					//check filters
-					if(ss.getSite() == null || !siteFilter.match(ss.getSite())) {
-						log.debug("Site with id={} skipped due to filter restrinctions", ss.getSiteId());
-						continue;
+				int retryCount = 0;
+				final int maxRetries = 3;
```
Comment (jesusmmp) [inline microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/jobs/RunSynchronizationsJob.java:85]:
```
private static final int MAXRETRIES = 3;
```

at the top.

File: microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/editGroupSynchronization.html [hunk 5]
```diff
@@ -26,22 +39,58 @@ <h1 class="header-title" th:text="#{group_synch.new_title}"></h1>
...
-						<div class="col-md-2 text-center">
-							<span th:text="#{delete}" class="text-bold"></span>
+						<div class="col-sm-2 text-center">
+							<span th:text="#{index.actions}" class="text-bold"></span>
 						</div>
 					</div>
 					<div id="row-container" class="row-list">
+						<div class="row table-space table-row">
+
+						<div class="col-sm-5">
```
Comment (jesusmmp) [inline microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/editGroupSynchronization.html:49]: Spacing

File: microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/editGroupSynchronization.html [hunk 6]
```diff
@@ -53,161 +102,140 @@ <h1 class="header-title" th:text="#{group_synch.new_title}"></h1>
...
-			const parentDiv = document.querySelector('#row-container');
-
-			parentDiv.addEventListener('click', handleButtonClick);
-			parentDiv.addEventListener('keypress', handleButtonClick);
+			        async function createGroupSynchronization(btn, url) {
+
+			        	let groupId = document.getElementById('selectedGroupId').value
+			        	let channelId = document.getElementById('selectedChannelId').value
+
+			        	if(!validateForm()) {
```
Comment (jesusmmp) [inline microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/editGroupSynchronization.html:114]:
```
if (!validateForm()) {
```

File: microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftCommonServiceImpl.java [hunk 7]
```diff
@@ -870,6 +1176,98 @@ public boolean addOwnerToTeam(String userId, String teamId) throws MicrosoftCred
...
+		boolean res = false;
+		String teamId = ss.getTeamId();
+		String dataKey = roles.contains(MicrosoftUser.OWNER) ? "ownerId" : "memberId";
+		boolean generalError = false;
+
+		ConversationMemberCollectionRequest postMembers = graphClient.teams(teamId).members()
+				.buildRequest();
+
+		final int MAX_RETRY = 2;
+		final int MAX_PER_REQUEST = 20;
```
Comment (jesusmmp) [inline microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftCommonServiceImpl.java:1190]: I'll prefer these at the top

**Comment by mollerentornos** [comment]
@jesusmmp PR updated with requested changes

**Review by jesusmmp — APPROVED** [review]

**Pull request merged**